service/paging.go

```diff
@@ -1,55 +1,74 @@
 package service
 
 import (
-	"fmt"
 	"net/http"
-	"strconv"
+	"strings"
 )
 
+type Sort struct {
+	Field     string
+	Direction string
+}
+
+func (s Sort) Valid(fieldWhiteList []string) bool {
+	if s.Field == "" || !contains(fieldWhiteList, s.Field) {
+		return false
+	}
+
+	switch s.Direction {
+	case "asc",
+		"desc":
+	// no-op
+	default:
+		return false
+	}
+
+	return true
+}
+
 // PagingParams represents paging and sorting parameter values
 type PagingParams struct {
 	PageSize   *int32
-	PageNumber *int32
-	SortField  *string
-	SortAsc    *bool
+	Offset     *int32
```

Comment:
We should add json tags to these so when we return the paging information we can use these structs. Also might be worth it to add the TotalCount param as well